57 add documentation

vanlankveldthijs opened the pull request:
To resolve issue #57

There are a few remaining issues to resolve.

1) The pull request has 2 example notebooks: one with and one without storing and reloading the ordered data before performing the timing tests. To be determined which is better. This one should be called `demo_order_stm.ipynb` and the other removed.

2) It seems like stmat.copy does not have the desired result: when I try to run STM operations on the cloned data, there seems to be some parts missing (I think it was the whole .stm part). Solved by loading the whole dataset twice, but that is a bit of an ugly fix.

3) It seems that when using small chunk sizes, storing to zarr fails. Solved by storing using 1000 size chunks and rechunking after loading. This is a bit ugly.

4) Possible issue with storing the ordered data to zarr when this needs to overwrite existing data. I think shutil.retree also sometimes gives issues (maybe when the is no directory).

5) Timing tests using the prerequisite small chunk size (500 in this case) become very slow. Don't know whether we can really fix this though.
Currently running the final timing tests on both notebooks. To be committed when done.

rogerkuou commented:
Thanks @vanlankveldthijs for the implementation. Following our discussion, the next steps can be:

1. Demonstrate the performance enhancement from re-ordering in a separate md page.
1. Elaborate on the factors affecting the re-ordering effect: 1) homogeneity of point distribution; 2) spatial spreadness of points. These two factors can influence the choice of chunksize and spatial closeness of points in the same chunk
1. Only do performance comparision on `subset`, which should be enough  
1. Use `%%timeit` for a robust profiling
1. Still keep an example notebook for the illustration of re-order operation (you can decide either include this in the existing notebooks or make a new one)

Attaching here two example notebooks I made for comparison: [notebooks.zip](https://github.com/TUDelftGeodesy/stmtools/files/14918049/notebooks.zip)

rogerkuou commented on .gitignore:
```diff
@@ -156,4 +156,10 @@ dmypy.json
\ No newline at end of file
+.vscode
+
+
+# Demo data
+docs/notebooks/bag_light_AMS_WGS84.gpkg
+docs/notebooks/stm.zarr
+docs/notebooks/stm_ordered.zarr
```
```suggestion
```
I would suggest let's not add these files to `.gitignore`. We do not have plans to host these example datasets in the repo.
IMO, for development purposes, I think people should be able to see the file they generated in git rather than ignore them.

rogerkuou reviewed (changes requested):
Hi @vanlankveldthijs, I also reviewed other changes you implemented. In general they are good. Just some minor comments:

1. I suggested a change on `.gitignore`;
2. The [build workflows](https://github.com/TUDelftGeodesy/stmtools/actions/runs/8599935945) are failing because the unit tests are filling. 
3. The [ruff workflow](https://github.com/TUDelftGeodesy/stmtools/actions/runs/8599935944/job/23563868034) failed because some linting problem. Sarah have already fixed this in #66. You do not need to worry about this.

vanlankveldthijs commented:
Consolidated demo notebooks into one.

Fixed stm.copy issue (should've been stm.copy())

Changed demo notebook to only test subset operation and to test this on large and small chunks (relative to dataset) to see impact.

vanlankveldthijs commented:
Fixed pytests by checking for existence of time in chunksizes.

Please check whether implementation could be nicer (better python).
